### DISCUSSION TITLE
Scripting Language - Berry Scripting Language differences

### DISCUSSION THREAD

**Discussion by nobodyman1** [arendst/Tasmota#18700] [Q&A]
I´m using an ESP32C3 controller with scripting.
In the documentation I found "Scripting Language for Tasmota is an alternative to Tasmota [Rules](https://tasmota.github.io/docs/Rules/). For ESP32 builds it is recommended to use [Berry](https://tasmota.github.io/docs/Berry/)."
But I think I have to use scripting combined with berry because some things don´t work with berry.

My berry example application:
```
class TEST
  var ctr

  def init()
    self.ctr = 0
  end

  def every_second()
    self.ctr += 1
  end

  def web_sensor()
    import string
    var msg = string.format(
        "{s}Ctr{m}%i{e}",            
        self.ctr)
     tasmota.web_send_decimal(msg)
  end
end

test = TEST()
tasmota.add_driver(test)
```

But the challenge is to have a button to enable or disable that driver.
Therefore I added a script (not berry)  to add this button.
```
>D
enable=0
>W
bu(enable "Berry script on" "Berry script off")
>F
if chg[enable]>0
then
if enable==1
then
print button 1
else
print button 0
endif
endif
#
```

If I click the button the console shows that the value is toggled as expected.
But I don´t know how to start or stop the berry script.

Any ideas?
Or is it possible to implement all in the berry code (which I would prefer)?

**Comment by sfromis** [comment]
With Berry, what you get using `web_sensor` is not a button. The Berry cookbook page has: [Adding a button to the Main menu](https://tasmota.github.io/docs/Berry-Cookbook/#adding-a-button-to-the-main-menu)

Not sure what you operationally mean by "But I don´t know how to start or stop the berry script", but simplest way to have Berry code run automatically after boot is via `autoexec.be`. And deactivating the code you posted would be by removing the driver.

**Comment by nobodyman1** [comment]
I know that using `web_sensor` doesn´t show a button. I try to explain it again.
I want to have a button in the WebUI. When I press the button I want to stop the running berry script.
The button text should switch between "on" and "off".
The berry script is an example which counts up a variable and shows the value in the WebUI.
If I manually remove the driver the counter stops. (But I want to use a button for this.)

**Reply by sfromis** [reply]
Well, read the link I posted about adding a button the the main menu.

**Comment by nobodyman1** [comment]
I tried the example but I want to change the button text to show the state.
I want to show "Script on" if it´s disabled and "Script off" if it´s enabled.
I have no idea how to implement this!

**Reply by sfromis** [reply]
Simplest is to not expect the state in the button, but as a separate value displayed.
